Addition of the first block of code to implement the registering of new experiments

aOlmo opened the pull request:
This Pull Request is made to add the first block of code that the previous contributor created while working in the functionality of registering of a new experiment. 

As a heads up, there are some parts in the PR code  that will be changed or erased in the next commits. To see which ones are and all the changes included in this PR, please, look at the last commit message. Finally, this PR may not work "as is" now but I consider it to be the fundamental base code for the future ones.

PS: I had to create a new PR because I had to change the commit messages and also some code, so please disregard the last PR I made.

Commit:
Add the first block of code to implement the registration of an experiment

- Add the new function create_experiment(): The main purpose of this function is
  to add a new experiment in the database.
- Add new functions to interface.py:
   - register_experiment: This function will add a new experiment to the database
     by using the methodcreate_experiment() in maindb.py
   - register_sensor: Function that adds a new sensor to the database calling the
     create_sensor method in maindb.py
- Add the following classes to models.py:
   - Experiment: Class that will save the basic data of a new experiment created
     by a researcher
   - Sensor: Abstract class where all sensors will inherit from. This saves the
     general data from each sensor: experiment_id, frequency, frequency_unit …
   - Battery, Bluetooth, Cellular, Location, Settings, ConcretSensor,  Signal_strengths,
     Wifi: All these classes are the ones that inherit from the Sensor class
     and will store their specific values.
- Add new forms RegisterExperimentForm, GeneralSensorAtributesForm and all sensor
   forms (BatteryForm, BluetoothForm, …) to the file website/html/forms.py
- Add more django blocks to website/html/templates/control/control_base.html
- Add file website/html/templates/control/registerexperiment.html This file is
  currently very improvable. In later commits this one is completely changed
  but for the moment it is necessary to have everything working together.
- Minor changes in indentation and spacing in website/html/urls.py
- Add registerexperiment function to website/html/views.py: This function
  registers a new experiment including all the sensors that a reasearcher
  might have chosen. This function although seems to work is very badly written
  and will be changed in the following commits.
- Changes in website/settings.py: Changes in the INSTALLER_BUILDER and addition
  of the new model: clearinghouse.website.control.models

aaaaalbert commented:
A few suggestions:
* Commit messages must be more descriptive than "Adding old code". Why is old code added? What does it do? (It adds `create_...` functions for users and sensors, right?)
* Following the above, if your commit message for 3f179c2 has all this information, then feel free to squash before sending the PR. See the Git docs for "rewriting history".
* Our [coding style guidelines](https://github.com/secure-systems-lab/code-style-guidelines) suggest two spaces per indent level. I see four in [some places](https://github.com/SensibilityTestbed/clearinghouse/pull/9/files#diff-1e03fcd439dd86f4ac50e3bd42ed55e5R290).
* Docstrings are incomplete, e.g. [here](https://github.com/SensibilityTestbed/clearinghouse/pull/9/files#diff-f0611a8f523174006215d3262e08bcd3R166) and [here](https://github.com/SensibilityTestbed/clearinghouse/pull/9/files#diff-f0611a8f523174006215d3262e08bcd3R207).
* Consider making `for field in [precision, goal, ...]: ...` loops out of repeated blocks like [these](https://github.com/SensibilityTestbed/clearinghouse/pull/9/files#diff-f0611a8f523174006215d3262e08bcd3R219).
 * [64 bytes for a street address](https://github.com/SensibilityTestbed/clearinghouse/pull/9/files#diff-ca9f89682718a695e0cb3a8c41ae55ecR139) is not too generous.
* Take care of maximum line lengths, e.g. [here](https://github.com/SensibilityTestbed/clearinghouse/pull/9/files#diff-ca9f89682718a695e0cb3a8c41ae55ecR139).
* Typo: ["precision"](https://github.com/SensibilityTestbed/clearinghouse/pull/9/files#diff-ca9f89682718a695e0cb3a8c41ae55ecR159).
* I don't understand the comments for the `..._other` properties of `Sensor`, such as [this](https://github.com/SensibilityTestbed/clearinghouse/pull/9/files#diff-ca9f89682718a695e0cb3a8c41ae55ecR171). We store what we don't support?
* Are [these `default` values](https://github.com/SensibilityTestbed/clearinghouse/pull/9/files#diff-ca9f89682718a695e0cb3a8c41ae55ecR197) ways of allowing or disallowing a sensor? Does the researcher request access to a sensor by providing `True`? The comments don't quite tell me.
* Typo: ["concrete"](https://github.com/SensibilityTestbed/clearinghouse/pull/9/files#diff-ca9f89682718a695e0cb3a8c41ae55ecR328).
* What about the many changes of file access bits? What's the purpose?

aOlmo commented:
Thanks a lot for the feedback, I will be working on all of this and will send another PR as soon as I have it. 
I just have a couple comments about what you mentioned:

In regards to this comment: 
> * Typo: "concrete".

This is solved later in another commit where I had to change several other things due to that renaming.

Sorry, I did not quite understand this last comment:
> * What about the many changes of file access bits? What's the purpose?

Thanks again for the feedback,

aaaaalbert commented:
> concret

Thanks, no problem.


> File access bits

See the "Files" view of the PR, https://github.com/SensibilityTestbed/clearinghouse/pull/9/files

Scroll down a bit until you see `common/util/nodestatus.py 100755 → 100644`. The number--arrow--number notation informs us that the access rights granted on that file have been changed. (See `man chmod` for reference). I wonder why that happened.

aOlmo committed:
Change indentation and access rights from 644 to 755

- Change indentation in all files inside "website" folder to be two spaces
  per each indentation level.
- Change in user bit rights from 644 to 755 to several files which were
  previously changed.

aOlmo commented:
**Here are the new changes in this PR:**

> * Commit messages must be more descriptive than "Adding old code". Why is old code added? What does it do? (It adds create_... functions for users and sensors, right?)
> * Following the above, if your commit message for 3f179c2 has all this information, then feel free to squash before sending the PR. See the Git docs for "rewriting history".

Those three last commits have been squashed into c382868 now. The commit message is the same as the last commit of those three.

>* Our coding style guidelines suggest two spaces per indent level. I see four in some places. 
>* What about the many changes of file access bits? What's the purpose?

Indentation has been changed to be 2 spaces per each indentation level and the file access bits are now the same as previously (I don't know why these were changed). This can be seen [here](https://github.com/aOlmo/clearinghouse/commit/84962a0a78ba291a5947bd1fac3577b8f1c2c783)

_________________________________________________________________________________________________

**Changes that have been made following the suggestions and will come in the next PRs:**

>* Docstrings are incomplete, e.g. here and here.

Those are now more complete [here](https://github.com/aOlmo/clearinghouse/commit/49314da685e700faed7a6df09ea487427c5135a6#diff-f0611a8f523174006215d3262e08bcd3R160) and [here](https://github.com/aOlmo/clearinghouse/commit/49314da685e700faed7a6df09ea487427c5135a6#diff-f0611a8f523174006215d3262e08bcd3R184) 

>* Consider making for field in [precision, goal, ...]: ... loops out of repeated blocks like these.

Done [here](https://github.com/aOlmo/clearinghouse/commit/b10eb25ff10a2cee09d9361be2d91251c5c86015#diff-f0611a8f523174006215d3262e08bcd3R209)

>* 64 bytes for a street address is not too generous.

In this [commit,](https://github.com/aOlmo/clearinghouse/commit/eda13414dfd00b11c787c0fba51469a6332667dc) all byte lengths have been reviewed and changed to the current standards following the second recommendation [here](https://stackoverflow.com/questions/20958/list-of-standard-lengths-for-database-fields) 

>* Take care of maximum line lengths, e.g. here.

Done in this [commit](https://github.com/aOlmo/clearinghouse/commit/b10eb25ff10a2cee09d9361be2d91251c5c86015)

>* Typo: "precision".

Corrected [here](https://github.com/aOlmo/clearinghouse/commit/f3632cd57317e452e611292e40045b3eb013f2e6#diff-ca9f89682718a695e0cb3a8c41ae55ecR162)

>* I don't understand the comments for the ..._other properties of Sensor, such as this. We store what we don't support?

The comments on "_other" properties of Sensor have been [rewritten](https://github.com/aOlmo/clearinghouse/commit/49314da685e700faed7a6df09ea487427c5135a6#diff-ca9f89682718a695e0cb3a8c41ae55ecR181)

>* Are these default values ways of allowing or disallowing a sensor? Does the researcher request access to a sensor by providing True? The comments don't quite tell me.

I have tried to explain it a little bit better in the docstrings from each class like the [Battery](https://github.com/aOlmo/clearinghouse/commit/49314da685e700faed7a6df09ea487427c5135a6#diff-ca9f89682718a695e0cb3a8c41ae55ecR209) one
